File: snippets/fsharp/System/UIntPtr/op_Addition/op_subtraction1.fs
module op_subtraction

open System

// <Snippet2>
let arr = [| 1; 2; 3; 4; 5; 6; 7; 8; 9; 10 |]
let ptr = UIntPtr(uint arr[arr.GetUpperBound 0])
for i = 0 to arr.GetUpperBound 0 do
    let newPtr = ptr - UIntPtr(uint i)
    printf $"{newPtr}   "
// </Snippet2>

File: snippets/csharp/System/UIntPtr/op_Addition/op_subtraction1.cs
using System;

public class UIntPtrSubtractionExample
{
    public static void Run()
    {
        // <Snippet2>
        int[] arr = { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 };
        UIntPtr ptr = (UIntPtr)arr[arr.GetUpperBound(0)];
        for (int ctr = 0; ctr <= arr.GetUpperBound(0); ctr++)
        {
            UIntPtr newPtr = ptr - (nuint)ctr;
            Console.Write($"{newPtr}   ");
        }
        // </Snippet2>
    }
}
